src/styles/text/text.js

```diff
@@ -92,15 +92,32 @@ Object.assign(TextStyle, {
             return;
         }
 
-        q.feature = feature;
-        q.context = context;
-        q.layout.vertex = false; // vertex placement option not applicable to standalone labels
+        // text can be an array if a `left` or `right` orientation key is defined for the text source
+        // in which case, push both text sources to the queue
+        if (q instanceof Array){
```

Comment:
If I were to choose, I'd choose `instanceof` for all array checking. The places where it doesn't work is for iframes... which, shouldn't be a problem for us?